File: browserfn/src/main/java/org/sqlunet/browser/fn/Application.java
/*
 * Copyright (c) 2023. Bernard Bou
 */

package org.sqlunet.browser.fn;

import android.content.Context;

import org.sqlunet.browser.AbstractApplication;
import org.sqlunet.style.Colors;

import androidx.annotation.NonNull;

public class Application extends AbstractApplication
{
	static private final String TAG = "Application";

	@Override
	public void onCreate()
	{
		super.onCreate();
		setAllColorsFromResources(this);
	}

	@Override
	public void setAllColorsFromResources(@NonNull final Context context)
	{
		// Log.d(TAG, "setColors " + NightMode.nightModeToString(this));
		Colors.setColorsFromResources(context);
		org.sqlunet.framenet.style.Colors.setColorsFromResources(context);
	}
}

File: browserfn/src/main/java/org/sqlunet/browser/fn/Application.kt
/*
 * Copyright (c) 2023. Bernard Bou
 */
package org.sqlunet.browser.fn

import android.content.Context
import org.sqlunet.browser.AbstractApplication
import org.sqlunet.browser.AppContext
import org.sqlunet.framenet.FnFlags
import org.sqlunet.framenet.style.Colors
import org.sqlunet.style.Colors.setColorsFromResources

class Application : AbstractApplication() {

    override fun onCreate() {
        super.onCreate()
        AppContext.init(this)
        setAllColorsFromResources(this)
    }

    override fun setAllColorsFromResources(newContext: Context) {
        // Log.d(TAG, "setColors " + NightMode.nightModeToString(this))
        setColorsFromResources(newContext)
        Colors.setColorsFromResources(newContext)
    }

    /**
     * Drop data
     *
     * @return true if flagged in build config
     */
    override fun dropData(): Boolean {
        return BuildConfig.DROP_DATA
    }


    /**
     * Build time
     *
     * @return build time
     */
    override fun buildTime(): String {
        return BuildConfig.BUILD_TIME
    }

    /**
     * Git hash
     *
     * @return git hasj
     */
    override fun gitHash(): String {
        return BuildConfig.GIT_HASH
    }

   companion object {
        init {
            FnFlags.standAlone = true
        }
    }
}
